### ISSUE TITLE
v19.08.2-3.0.3 - Error code 9 during Actions

### ISSUE THREAD

**Issue by Scope666** [qnblackcat/uYouPlus#1469]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Have you read the FAQ?

- [X] Yes, I have read the FAQ

### Device information

- iOS/iPadOS version: latest
- Device model: 15 Pro Max
- Sideload tool (AltStore, Sideloadly, TrollStore, etc.): AltStore

### Describe the issue

Action can't complete.

### Steps to reproduce

Action can't grab the decrypted IPA even though it's available at the link.

### Crash log

```shell
THEOS: /Users/runner/work/uYouPlus/uYouPlus/theos
YOUTUBE_URL: https://drive.google.com/file/d/10bII1rwSudpoGQPTtBHOzTUh3aYg8c0x/view?usp=sharing
2024-05-21 15:36:07 URL:https://drive.google.com/file/d/10bII1rwSudpoGQPTtBHOzxxxxxxxxxx/view?usp=sharing [85244] -> "main/YouTube.ipa" [1]
[YouTube.zip]
End-of-central-directory signature not found. Either this file is not
a zipfile, or it constitutes one disk of a multi-part archive. In the
latter case the central directory and zipfile comment will be found on
the last disk(s) of this archive.
unzip: cannot find zipfile directory in one of YouTube.zip or
YouTube.zip.zip, and cannot find YouTube.zip.ZIP, period.
Error: Process completed with exit code 9.
```

### Are you using the latest version of uYouPlus? If not, why?

✅ Yes, I'm using the latest version of uYouPlus

### Does the issue happen with the official YouTube app from the App Store?

❌ No, I cannot reproduce this issue in the official YouTube app from the App Store

### Additional information

Error: Process completed with exit code 9.

**Comment by qnblackcat** [comment]
Are you sure you provided a direct URL?

**Comment by Scope666** [comment]
Yes, if you click the Drive link it downloads the .IPA.

**Comment by Scope666** [comment]
I even tried this, it seems to be a problem with Google Drive.  https://sites.google.com/site/gdocs2direct/

I just tried Dropbox and that built successfully.

**Comment by Lifell** [comment]
You need a direct-direct link to make Google Drive work. That is not the sharing link that google drive provides. Took me awhile to figure this out ._.

**Comment by harshilkumar-jayswal** [comment]
Yes , I was getting the same error with my build. If you are also noob, here are some useful instructions for dropbox. https://zapier.com/blog/generate-direct-dropbox-link/

**Comment by Lifell** [comment]
> Yes , I was getting the same error with my build. If you are also a noob like me, here are some useful instructions for dropbox. https://zapier.com/blog/generate-direct-dropbox-link/

I managed to get it to work with Google Drive yesterday. My download manager could provide me with an direct link. Thanks for the link though!

**Comment by EdwinKuo1337** [comment]
Just to add more information to this, it is because the file is too large, so when you paste the direct download link for google drive you get from https://sites.google.com/site/gdocs2direct/, it actually opens a page saying that the file is too large for Google to scan for viruses. That page will have a button to download the file. So the direct download link you actually want is the url for that button instead of the link `gdocs2direct` gives you

**Comment by thanhphong97** [comment]
> Just to add more information to this, it is because the file is too large, so when you paste the direct download link for google drive you get from https://sites.google.com/site/gdocs2direct/, it actually opens a page saying that the file is too large for Google to scan for viruses. That page will have a button to download the file.
> 
> How to get the actual direct download link: before you click the download button, go to inspect element -> network tab, then click direct download button. In the network tab you should see a request. Click on that request and you should see a Request URL, and that is your actual direct down link!
Thanks, you are a life saver
